### ISSUE TITLE
Longer text can't fit in <input> elements of the types button, submit and reset 

### ISSUE THREAD

**Issue by biljanap** [wet-boew/wet-boew#7744]
The `<input>` elements of the types button, submit and reset don’t handle well longer button text because they can only display one line of text. This is caused by fixed css height.
Any text that goes on the second line becomes invisible.

![input-btn-long-text](https://cloud.githubusercontent.com/assets/2146304/20072006/bcb3c5fc-a4f4-11e6-9538-1f19ecde36f2.jpg)

The first button in the example is `<button>` element. The second button is `<input`> element.
HTML code:
```html
<button class="btn btn-default" type="submit">Longer text on button element</button>
<input class="btn btn-default" value="Longer text on input element" type="button">
```
css code (theme.min.css):
```html
input[type="button"], input[type="reset"], input[type="submit"] {
    height: 37px;
}
```
Is it possible to remove the fixed height?

**Comment by masterbee** [comment]
@biljanap - thanks for this. I am just curious is this the WET base  or specific to a theme in WET. GCWeb perhaps?

**Comment by biljanap** [comment]
@masterbee . I am using Canada.ca theme. I did not check anything else.

**Comment by biljanap** [comment]
@masterbee this is the case with WET base as well. Just checked.

**Comment by LaurentGoderre** [comment]
That style comes from Bootstrap.

**Comment by sgdowney** [comment]
Is there any preference for using a `<button>` over an `<input type="submit">`?

**Comment by RobJohnston** [comment]
@sgdowney , I remember one reason from the [style guide] (http://wet-boew.github.io/wet-boew-styleguide/v4/design/buttons-en.html):

> Use the `<button>` element whenever possible to ensure matching cross-browser rendering
> There is a Firefox bug that makes it impossible to set the` line-height` in `<input>`-based buttons
> These buttons therefore have uneven heights in Firefox

**Comment by kodecount** [comment]
@RobJohnston I looks like the bug is fixed.

**Comment by TkTech** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=697451

Resolved in 2014

**Comment by RobJohnston** [comment]
Then maybe somebody should update the style guide to mention that it was fixed for Firefox 30+.  I'm not aware of any other reason to prefer `<button>` over `<input>`, but others may be.

**Comment by kodecount** [comment]
> Buttons created with the BUTTON element function just like buttons created with the INPUT element, but they offer richer rendering possibilities: the BUTTON element may have content. For example, a BUTTON element that contains an image functions like and may resemble an INPUT element whose type is set to "image", but the BUTTON element type allows content. - https://www.w3.org/TR/html4/interact/forms.html#h-17.5

**Comment by LaurentGoderre** [comment]
Here's what Bootstrap is recommending http://getbootstrap.com/css/#callout-buttons-ff-height

**Comment by RobJohnston** [comment]
Exactly the wording in the style guide that I referenced, which now appears now to be outdated.

**Comment by biljanap** [comment]
Adding fixed height to `<input>` might have been a necessary solution to Firefox `line-height` bug but since this bug has been fixed since I believe it would be safe to just remove the fixed height from `<input>`.

@sgdowney We don't have preference for using `<button>` over an `<input type="submit">`. We are converting an existing app to WET4 and `<input type="submit">` was inherited. From everything I researched to see if it is safe to replace `<input type="submit">` with `<button>` it seems they both do the same thing but we have more flexibility over styling `<button>` element. If this is only difference, it's better to use `<button>` if support for older browsers (IE6/7) is not an issue.

**Comment by kodecount** [comment]
@biljanap Well WET 4 is only supporting IE10 and higher. It works okay on IE 8 and 9, but anything lower it don't.

**Comment by kodecount** [comment]
@biljanap Well WET 4 is only supporting IE10 and higher. It works okay on IE 8 and 9, but anything lower it don't.

Also as @LaurentGoderre has mention this is set within bootstrap. WET would need to override it with `height: auto !important;` which be messy. Just easier to allow anyone using WET to add this to their own style-sheets.

**Comment by biljanap** [comment]
@kodecount fixed height comes from theme.css. This is not Bootstrap, right?
```html
input[type="reset"], input[type="button"], input[type="submit"] {
  height: 37px; }
```
I am not finding these styles in Bootstrap.
Also from what I can see both `<input type="button">` and `<button>` elements render differently on Bootstrap web site. The text does not wrap onto the next line. Both element render as very long buttons exceeding the borders of their container.

**Comment by kodecount** [comment]
@biljanap All the code is generated behind the scenes using nodejs and sass.

**Comment by biljanap** [comment]
Strange then that Bootstrap `<input type="button">` behaves differently (no wrapping). I have bootstrap-3.3.7-dist downloaded and I am not finding these styles. Unless this happens in some of Bootstrap themes.
Also we are moving our content to AEM and it's possible we won't be able to use custom css. Mario 
@masterbee might know.

**Comment by kodecount** [comment]
WET is still on version 3.3.1 of bootstrap. The bootstrap sass library I believe is updated to 3.3.7 but someone at WET would have to make the change to update WET to latest stable bootstrap and wouldn't be released until the next version of WET.

**Comment by kodecount** [comment]
@masterbee Is it in the works to update bootstrap too version 3.3.7?

**Comment by duboisp** [comment]
FYI - The latest released version of WET 4.0.22 and the next one v4.0.23 are using [Boostrap v3.1.1](https://github.com/wet-boew/wet-boew/blob/master/bower.json#L25)

**Comment by kodecount** [comment]
@duboisp it 3.3.1 and is there any chance of updating to 3.3.7?

**Comment by duboisp** [comment]
@kodecount yes it is possible but it would requires to test every plugin and functionality to ensure they are not going to be broken somehow. Also, it would required to check what has been changed since 3.1.1 and resolve any issue regarding our own styling. That would also require to test all the supported WET theme and fix them if required.

**Comment by kodecount** [comment]
@biljanap @masterbee I believe the issue be closed. The user well be required to use a `<button>` or override the styling until/if bootstrap is updated in the future.

**Comment by shawnthompson** [comment]
I just reopened this issue. This is a problem when resizing the text at 200%

https://github.com/wet-boew/wet-boew/issues/6910

**Comment by biljanap** [comment]
Are you choosing "Zoom > Zoom Text Only"? If yes, the input element would not zoom.

**Comment by biljanap** [comment]
....and this would the case with any ```<input>```. The text would be cut.

**Comment by thekodester** [comment]
@shawnthompson This issue was not about a ZOOM issue, but was about the fact `<input/>` buttons could not handle long strings of text.

**Comment by shawnthompson** [comment]
@thekodester here's a working copy of `<input type="button" value="long text" />` actually wrapping.

https://shawnthompson.github.io/themes-dist-4.0.27-gcweb/input_button.html

I just added the following:

```css
    input#button {
        white-space: normal;
        height: auto;
    }
```
In the theme.css there's this rule that sets the height:

```css
input[type="button"], input[type="reset"], input[type="submit"] {
	height: 37px;
}
```
